src/abstractdataframe/join.jl

```diff
@@ -337,10 +352,27 @@ function Base.join(df1::AbstractDataFrame,
                 push!(leftonly_ixs, l_ix)
             end
         end
-        return joiner.dfl[leftonly_ixs, :]
+        joined = joiner.dfl[leftonly_ixs, :]
     else
         throw(ArgumentError("Unknown kind of join requested: $kind"))
     end
+
+    if !(indicator == nothing)
+        left = joined[indicator_cols[1]]
+        left = Int.(coalesce.(left, false))
+
+        right = joined[indicator_cols[2]]
+        right = Int.(coalesce.(right, false)) * 2
+
+        _merge = DataFrame(indicator => categorical(left + right))
+        _merge[indicator] = recode(_merge[indicator], 1 => "left_only", 2 => "right_only", 3 => "both")
+        joined = hcat(joined, _merge, makeunique=true)
```

Comment:
This is inconsistent with the documentation, where you state that you use the value of `makeunique` passed by the user. Here you override it by setting it to `true`. I thought you want to use what the user has passed (this is probably not relevant in interactive use but is relevant when used programmatically).

Reply:
OH! I didn't mean to suggest I'd obey the user's argument passed to `makeunique` -- I meant that renaming takes place using `makeunique` norms (i.e. the way it updates column names when a conflict occurs is using `makeunique`). Will fix docs.

Reply:
Actually I agreed with the docs. Why it is important in my opinion to be consistent with what user passes in `makeunique` is the following scenario.

Assume we are using `indicator` keyword argument in a program (not interactive use). A program then wants to access the column named after `indicator`. If you always silently rename the column you may run into problems trying to access the wrong column. It is better to throw an error if `makeunique` is set to `false` by the user in such a case. The user can always pass `makeunique=ture` if she wants column to be renamed.

Reply:
Right, super fair. Will adapt in that direction then!